ChronoVisor/src/ChronicleMetaDirectory.cpp

```diff
@@ -108,61 +138,87 @@ bool ChronicleMetaDirectory::create_story(std::string &chronicle_name,
         Chronicle *pChronicle = chronicleRecord->second;
         LOGD("Chronicle@%p", &(*pChronicle));
         bool res = pChronicle->addStory(chronicle_name, story_name, attrs);
```

Comment:
What if the story already exists in the chronicle map? You have to add it only once and fix the memory leak if it is a duplicate.